components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -100,14 +110,33 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
 
   closeDropdown(): void {
     this.dropdownState = 'closed';
-    this.onDropdownClosing.emit(
-      this.resources.filter(r => r.checked).map(r => r.id));
+    const flattenedIDs = [].concat(
+      ...this.snapshotResources.map(
+        resource => resource.itemList.filter(r => r.checked).map(r => r.id)));
+    this.onDropdownClosing.emit(flattenedIDs);
   }
 
   handleFilterKeyUp(): void {
-    this.filteredResources = this.resourcesInOrder.filter(resource =>
-      resource.id.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1
-    );
+    for (let i = 0; i < this.filteredResources.length; i++) {
+      this.filteredResources[i].itemList =
+        this.snapshotResources[i].itemList.filter(r =>
+          r.name.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1);
+    }
+    this.allFilteredResourcesCount = this.calculateAllFilteredResourcesCount();
+  }
+
+  private resetFilteredResources(): void {
+
+    // Not deep and not shallow!
+    // Need the individual resources to point to the same objects
+    // so that when one is checked, it is checked in BOTH structures.
+    this.filteredResources = [].concat(
```

Comment:
Same as my question above :)

Reply:
...has it to do with the specific type of copy you want there? 🤔

Reply:
Nothing esoteric; changed this to flatMap, too. 👍